components/cam/src/physics/cam/shoc.F90

```diff
@@ -3102,32 +3173,123 @@ real(rtype) :: phihinv(shcol)          ! inverse phi function for heat
+
+    !
+    ! Output arguments
+    !
+    real(rtype), intent(out)   :: pblh(shcol)             ! boundary-layer height [m]
+    real(rtype), intent(inout) :: rino(shcol,nlev)                     ! bulk Richardson no. from level to ref lev
+    logical, intent(inout)     :: check(shcol)            ! True=>chk if Richardson no.>critcal
+
+    !
+    !---------------------------Local workspace-----------------------------
     !
+    integer  :: i                       ! longitude index
+    integer  :: k                       ! level index
+    real(rtype) :: vvk                     ! velocity magnitude squared
+    real(rtype) :: th
     !
     ! PBL height calculation:  Scan upward until the Richardson number between
     ! the first level and the current level exceeds the "critical" value.
     !
     do k=nlev-1,nlev-npbl+1,-1
        do i=1,shcol
           if (check(i)) then
-             vvk = (u(i,k) - u(i,nlev))**2 + (v(i,k) - v(i,nlev))**2 + fac*ustar(i)**2
+             vvk = (u(i,k) - u(i,nlev))**2 + (v(i,k) - v(i,nlev))**2 +fac*ustar(i)**2
              vvk = max(vvk,tiny)
-             rino(i,k) = ggr*(thv(i,k) - thv(i,nlev))*(z(i,k)-z(i,nlev))/(thv(i,nlev)*vvk)
+             rino(i,k) = ggr*(thv(i,k) -thv(i,nlev))*(z(i,k)-z(i,nlev))/(thv(i,nlev)*vvk)
              if (rino(i,k) >= ricr) then
-                pblh(i) = z(i,k+1) + (ricr - rino(i,k+1))/(rino(i,k) - rino(i,k+1)) * &
+                pblh(i) = z(i,k+1) + (ricr - rino(i,k+1))/(rino(i,k) -rino(i,k+1)) * &
                      (z(i,k) - z(i,k+1))
                 check(i) = .false.
              end if
           end if
        end do
     end do
+    return
+end subroutine pblintd_height
+
+subroutine pblintd_surf_temp(&
+       shcol,nlev,nlevi,&          ! Input
+       z,ustar,obklen,kbfs,thv,&   ! Input
+       phiminv,tlv,unstbl,&        ! Output
+       pblh,check,rino)            ! InOutput
+    !------------------------------Arguments--------------------------------
+    ! Input arguments
+    !
+    integer, intent(in) :: shcol                     ! number of atmospheric columns
+    integer, intent(in) :: nlev                      ! number of mid-point layers
+    integer, intent(in) :: nlevi                     ! number of interface layers
+
+    real(rtype), intent(in)  :: z(shcol,nlev)           ! height above surface [m]
+    real(rtype), intent(in)  :: ustar(shcol)            ! surface friction velocity [m/s]
+    real(rtype), intent(in)  :: obklen(shcol)           ! Obukhov length
+    real(rtype), intent(in)  :: kbfs(shcol)             ! sfc kinematic buoyancy flux [m^2/s^3]
+    real(rtype), intent(in) :: thv(shcol,nlev)          ! virtual potential temperature
+
+    real(rtype), intent(out) :: phiminv(shcol)          ! inverse phi function for momentum
+    real(rtype), intent(out) :: tlv(shcol)              ! ref. level pot tmp + tmp excess
+    logical, intent(out)  :: unstbl(shcol)              ! pts w/unstbl pbl (positive virtual ht flx)
```

Comment:
@bogensch might know more about it but if I am reading the code correctly, `unstbl` variable is being computed but is not used anywhere in the code. It might still be useful to keep this variable.